lib/guardian/topic_guardian.rb

```diff
@@ -130,9 +133,11 @@ def can_edit_topic?(topic)
     )
 
     return false if topic.archived
+
     is_my_own?(topic) &&
       !topic.edit_time_limit_expired?(user) &&
-      !Post.where(topic_id: topic.id, post_number: 1).where.not(locked_by_id: nil).exists?
+      !first_post&.locked? &&
+      (!first_post&.hidden? || can_edit_hidden_post?(first_post))
```

Comment:
Does this work? The `can_edit_hidden_post?` part, here in `TopicGuardian`

Also, I'm always uneasy about using `!` and `&.` together. e.g. `!first_post&.locked?` is true when the post is unlocked or when there's no post. Is that ok in these cases?

Reply:
> Does this work? The `can_edit_hidden_post?` part, here in `TopicGuardian`

Yes, it does because we know the `Guardian` class includes both the `TopicGuardian`, and the `PostGuardian` modules, making them available as instance methods. We always make use of these methods through the `Guardian` class.

> Also, I'm always uneasy about using ! and &. together. e.g. !first_post&.locked? is true when the post is unlocked or when there's no post. Is that ok in these cases?

The part I'm replacing also evaluates to true if there's no first post, so it's equivalent.